File: Matlab/SCRoF_Fevrier/Doubl.m
function coPlus=Doubl(coPlus)
% coPlus=Doubl(coPlus);
% Chaque ligne de coPlus contient une probabilité, trois rangs de grappes et un 0 ou code
d4=find(coPlus(:,end)==4,1,'last');
coP=coPlus(coPlus(:,end)==0,2:4);
cens=recens(coP);
d=0;
while any(cens(:,2)>1)
    f=find(coPlus(:,end)==0);
    r=find(cens(:,2)>1,1);
    r=cens(r,1);
    li=lignesAvec(coP,r);
    L=[];
    g=coP(li(1),:);
    if numel(li)>2
        keyboard % si ça arrive, l'utilisation de li comme pointeur plus bas pourrait être incorrecte
    end
    for k=2:numel(li)   % numel(li) n'est-il pas forcément 2?
        uni=unique([g,coP(li(k),:)]);
        if numel(uni)==5
            L=[f(1) f(k) 2+d];
        end
        if numel(uni)==4
            L=[f(1) f(k) 3+d];
        end
    end
    coPlus(li+d4,end)=L(3);
    d=d+10;
    coP=coPlus(coPlus(:,end)==0,2:4);
    cens=recens(coP);
end

File: R0/Doubl.R
Doubl <- function(coPlus) {
  # coPlus=Doubl(coPlus);
  # Chaque ligne de coPlus contient une probabilité, trois rangs de grappes et un 0 ou code
  
  d4 <- tail(which(coPlus[, ncol(coPlus)] == 4), 1)  # Équivalent de find(..., 'last')
  coP <- coPlus[coPlus[, ncol(coPlus)] == 0, 2:4]
  cens <- recens(coP)
  d <- 0
  
  while (any(cens[, 2] > 1)) {
    f <- which(coPlus[, ncol(coPlus)] == 0)
    r <- cens[which(cens[, 2] > 1)[1], 1]
    li <- lignesAvec(coP, r)
    L <- integer()
    g <- coP[li[1], ]
    
    if (length(li) > 2) {
      browser()  # Remplace 'keyboard' en MATLAB pour le debug en R
    }
    
    for (k in 2:length(li)) {  # length(li) au lieu de numel(li)
      uni <- unique(c(g, coP[li[k], ]))
      if (length(uni) == 5) {
        L <- c(f[1], f[k], 2 + d)
      }
      if (length(uni) == 4) {
        L <- c(f[1], f[k], 3 + d)
      }
    }
    
    coPlus[li + d4, ncol(coPlus)] <- L[3]
    d <- d + 10
    
    coP <- coPlus[coPlus[, ncol(coPlus)] == 0, 2:4]
    cens <- recens(coP)
  }
  
  return(coPlus)
}